docs/RFCs/build-process.md

```diff
@@ -0,0 +1,86 @@
+# Building process proposal
+
+## Context
+As described in the [RFC](https://github.com/DataDog/libdatadog/blob/main/docs/RFCs/delivering-different-features.md)
+about delivering different features the current build process depends heavily on shell scripts which makes it too 
+difficult in terms of managning the process itself like handling errors, dependendencies and features. Also shell
+scripts by themselves poses some disadvantages like portability and maintanibilty.
+
+This solution aims to get rid of those problems so the whole procedure depends as much as possible of built-in features 
+of cargo making it less error prone and also improving its maintanibilty.
+
+## Possible solutions
+Since we need some sort of post build actions and cargo currently does not provide any built-in method to carry out these 
+kind of actions we evaluated several options:
+
+### Add a new crate to manage the build process
+Add another crate in the workspace which will manage the postbuild actions and generate the artifacts. 
+
+### Use cargo-make
+cargo-make enables to define and configure sets of tasks and run them as a flow. A task is a command, script, rust
+code, or other sub tasks to execute. Tasks can have dependencies which are also tasks that will be executed before the
+task itself.
+
+### Use cargo-post
+A cargo wrapper which runs a post build script whenever the current build runs successfully.
+
+## Proposed solution
```

Comment:
I've noticed in other projects a similar "trope" 

https://github.com/cross-rs/cross/blob/main/xtask/src/main.rs

They define a custom https://github.com/cross-rs/cross/blob/main/.cargo/config.toml alias. That essentially makes a local crate execute a custom cargo command:

This way - they get things like:

```
cargo xtask changelog
cargo xtask test
```
as the main interface for interacting with the repository that is shared both with CI and local dev workflow.

IMO this looks neat and avoid having to deal with additional limitations of Makefiles etc. 

it can also be used together with build.rs 

But to me it would be good to have common tooling that can be executed explicitly at will.

Reply:
Agree, when I reviewed cargo-post I thought exactly the same. Once the build system is in place we can tackle this.